tracardi/process_engine/action/v1/operations/contains_pattern/plugin.py

```diff
@@ -0,0 +1,152 @@
+from pydantic import validator
+import re
+
+from tracardi.process_engine.action.v1.operations.contains_pattern.patterns import patterns
+from tracardi.service.plugin.domain.register import Plugin, Spec, MetaData, Documentation, PortDoc, Form, FormGroup, \
+    FormField, FormComponent
+from tracardi.service.plugin.runner import ActionRunner
+from tracardi.service.plugin.domain.config import PluginConfig
+from tracardi.service.plugin.domain.result import Result
+
+
+class Config(PluginConfig):
+    field: str
+    pattern: str
+
+    @validator("field")
+    def validate_prefix(cls, value):
+        if value == "":
+            raise ValueError("Field cannot be empty")
+        return value
+
+    @validator("pattern")
+    def validate_field(cls, value):
+        if value == "":
+            raise ValueError("Pattern cannot be empty")
+        return value
+
+
+def validate(config: dict) -> Config:
+    return Config(**config)
+
+
+class WrongFieldTypeError(Exception):
+    """Raised when given field has wrong data type"""
+    pass
+
+
+class EmptyPatternError(Exception):
+    """Raised when pattern field is empty"""
+    pass
+
+
+class ContainsPatternAction(ActionRunner):
+    config: Config
+
+    async def set_up(self, init):
+        self.config = validate(init)
+
+    async def run(self, payload: dict, in_edge=None):
+        dot = self._get_dot_accessor(payload)
+        value = dot[self.config.field]
+
+        if not isinstance(value, str) or isinstance(value, list):
+            raise WrongFieldTypeError(f"Given field must be an array or string type. {type(value)} given")
+
+
+        founded_patterns = []
+        if self.config.pattern == "all":
+            for k, v in patterns.items():
+                if re.findall(patterns[k], value):
+                    founded_patterns.append(k)
+            return Result(port='true', value={"found": founded_patterns})
+
+        if self.config.pattern == "email":
+            if re.match(patterns["email"], value):
+                return Result(port="true", value={"found": "email"})
+
+        if self.config.pattern == "url":
+            if re.match(patterns["url"], value):
+                return Result(port="true", value={"found": "url"})
+
+        if self.config.pattern == "date":
+            if re.match(patterns["date"], value):
+                return Result(port="true", value={"found": "date"})
+
+        if self.config.pattern == "ip":
+            if re.match(patterns["ip"], value):
+                return Result(port="true", value={"found": "ip"})
+
+        return Result(port="false", value="Didn't find any patterns")
```

Comment:
do not return string as value. That is going to raise an error in the further execution. Return something like value={"found": None} or {"message": "Did not find any pattern"}.